### PR TITLE
Shoatman/unified/dev/1057

### PR DESCRIPTION
**Pull request by shoatman**
Update ADAL to leverage common core for caching.  Changes include:

- TokenCacheAccessor - Update to use common cache when saving tokens (based on whether DefaultTokenCache was in use)
- Updates to tests to fix issues related to when Authority Host / Cloud Metadata data was loaded
- Updates to tests to include raw id_tokens to ensure proper behavior in common (TODO: Add support for ADFS.... we may only want to support common cache in the case of AAD)

### CODE REVIEW

File: adal/src/androidTest/java/com/microsoft/aad/adal/AuthenticationContextTest.java [hunk 1]
```diff
@@ -110,7 +114,7 @@
      */
     private static final String VALID_AUTHORITY = "https://login.windows.net/test.onmicrosoft.com";
 
-    protected static final int CONTEXT_REQUEST_TIME_OUT = 20000;
+    protected static final int CONTEXT_REQUEST_TIME_OUT = 20000; //20000
```
Comment (iambmelt) [inline adal/src/androidTest/java/com/microsoft/aad/adal/AuthenticationContextTest.java]: nit - stray comment

**Comment by iambmelt** [comment]
We may also want to create an issue for ADFS support since that appears to remain  a TODO item

File: adal/src/androidTest/java/com/microsoft/aad/adal/AuthenticationContextTest.java (hunk 1 above)
Comment (shoatman) [reply adal/src/androidTest/java/com/microsoft/aad/adal/AuthenticationContextTest.java]: I expanded on this comment to provide a tip for future developers.  Thanks for the reminder.

**Comment by shoatman** [comment]
I added an issue to address ADFS scenario.  See #1067

**Pull request merged**